iOS/CodePushPackage.m

```diff
@@ -288,18 +290,24 @@ + (void)downloadPackage:(NSDictionary *)updatePackage
                     NSArray *deletedFiles = manifestJSON[@"deletedFiles"];
                     for (NSString *deletedFileName in deletedFiles) {
                         [[NSFileManager defaultManager] removeItemAtPath:[newPackageFolderPath stringByAppendingPathComponent:deletedFileName]
-                                                                   error:&nonFailingError];
-                        
-                        if (nonFailingError) {
-                            NSLog(@"Error deleting file from current package: %@", nonFailingError);
-                            nonFailingError = nil;
+                                                                   error:&error];
+                        if (error) {
+                            failCallback(error);
```

Comment:
This will not.. i think we should not. However, maybe I can tweak this slightly to check whether the file exists before deleting it.

1. In the event the file doesn't exist, we should just carry on, because we achieved the effect we want of ensuring the file is removed. The fact that we received a manifest asking us to delete a file that doesnt exist is suspect, but so long as the final hash matches, I think we're good.

2. In the event an IO error occurs in deleting it, we should not blacklist it because it is not a problem with the update.